packages/core/strapi/lib/services/entity-service/index.d.ts

```diff
@@ -19,8 +21,12 @@ type PaginationInfo = {
   total: number;
 };
 
-type Params<T> = {
-  fields?: (keyof T)[];
+type DataParam<TSchemaUID extends Common.UID.Schema> = {
+  [Key in Attribute.GetKeys<TSchemaUID>]?: AttributeUtils.GetValue<Attribute.Get<TSchemaUID, Key>>;
```

Comment:
What about required fields?
Shouldn't they be required in here too?

Reply:
I'm wondering if we could find a better name for this, maybe something like Payload? Or Input?

Reply:
```suggestion
  [TKey in Attribute.GetKeys<TSchemaUID>]?: AttributeUtils.GetValue<Attribute.Get<TSchemaUID, TKey>>;
```